<launch>
  <arg name="camera_name" default="sim_camera1" />
  <group ns="$(arg camera_name)" >
    <param name="~camera_name" value="$(arg camera_name)" type="str"/>
    <node pkg="mct_simulation" type="sim_camera_render.py" name="sim_camera_render"></node>
    <node pkg="mct_simulation" type="sim_camera_driver" name="sim_camera_driver">
      <param name="camera_info_url" value="package://mct_config/calibration/calibration_$(arg camera_name).yaml" />
    </node>
    <node ns="camera" pkg="image_proc" type="image_proc" name="image_proc" />
    <node pkg="mct_simulation" type="render_tf_broadcaster.py" name="render_tf_broadcaster" />
  </group>

  <arg name="origin_camera_name" default="sim_camera6" />
  <group ns="$(arg origin_camera_name)" >
    <param name="~camera_name" value="$(arg origin_camera_name)" type="str"/>
    <node pkg="mct_simulation" type="sim_camera_render.py" name="sim_camera_render"></node>
    <node pkg="mct_simulation" type="sim_camera_driver" name="sim_camera_driver">
      <param name="camera_info_url" value="package://mct_config/calibration/calibration_$(arg origin_camera_name).yaml" />
    </node>
    <node ns="camera" pkg="image_proc" type="image_proc" name="image_proc" />
  </group>

  <include file="$(find camera_pose_calibration)/blocks/calibration_tf_publisher.launch">
    <arg name="cache_file" value="/tmp/camera_pose_calibration_cache.bag" />
  </include>

  <!-- <arg name="camera1_ns" value="$/(arg origin_camera_name)/camera/"/> -->
  <!-- <arg name="camera2_ns" value="$/(arg camera_name)/camera/"/> -->
  <!-- <arg name="checker_rows" value="7"/> -->
  <!-- <arg name="checker_cols" value="5"/> -->
  <!-- <arg name="checker_size" value="0.127"/> -->
  <!-- <include file="$(find camera_pose_calibration)/launch_extrinsics/calibrate_2_camera.launch" /> -->

</launch>
